test/liquid/combinators/tags/for_test.exs

```diff
@@ -0,0 +1,217 @@
+        for: [
+          for_collection: [
+            variable_name: "item",
+            value: {:variable, ["array"]}
+          ],
+          for_body: [],
+          else: []
+        ]
+      ])
+    end)
+  end
+
+  test "for tag: limit parameter" do
+    tags = [
+      "{% for item in array limit:2 %}{% else %}{% endfor %}",
+      "{%for item in array limit:2%}{%else%}{%endfor%}",
+      "{%     for     item    in     array  limit:2  %}{%   else    %}{%    endfor    %}",
+      "{%     for     item    in     array  limit: 2  %}{%   else    %}{%    endfor    %}"
+    ]
+
+    Enum.each(tags, fn tag ->
+      test_combinator(tag, &Parser.for/1,  [
+        for: [
+          for_collection: [
+            variable_name: "item",
+            value: {:variable, ["array"]},
+            limit_param: [2]
+          ],
+          for_body: [],
+          else: []
+        ]
+      ])
+    end)
+  end
+
+  test "for tag: offset parameter" do
+    tags = [
+      "{% for item in array offset:2 %}{% else %}{% endfor %}",
+      "{%for item in array offset:2%}{%else%}{%endfor%}",
+      "{%     for     item    in     array  offset:2  %}{%   else    %}{%    endfor    %}"
+    ]
+
+    Enum.each(tags, fn tag ->
+      test_combinator(tag, &Parser.for/1, [
+        for: [
+          for_collection: [
+            variable_name: "item",
+            value: {:variable, ["array"]},
+            offset_param: [2]
+          ],
+          for_body: [],
+          else: []
+        ]
+      ])
+    end)
+  end
+
+  test "for tag: reversed parameter" do
+    tags = [
+      "{% for item in array reversed %}{% else %}{% endfor %}",
+      "{%for item in array reversed%}{%else%}{%endfor%}",
+      "{%     for     item    in     array  reversed  %}{%   else    %}{%    endfor    %}"
+    ]
+
+    Enum.each(tags, fn tag ->
+      test_combinator(tag, &Parser.for/1, [
+        for: [
+          for_collection: [
+            variable_name: "item",
+            value: {:variable, ["array"]},
+            reversed_param: []
+          ],
+          for_body: [],
+          else: []
+        ]
+      ])
+    end)
+  end
+
+  test "for tag: range parameter" do
+    tags = [
+      "{% for i in (1..10) %}{{ i }}{% endfor %}",
+      "{%for i in (1..10)%}{{ i }}{% endfor %}",
+      "{%     for     i     in     (1..10)      %}{{ i }}{%     endfor     %}"
+    ]
+
+    Enum.each(tags, fn tag ->
+      test_combinator(
+        tag,
+        &Parser.for/1,
+        for: [
+          for_collection: [
+            variable_name: "i",
+            range_value: [start: 1, end: 10]
+          ],
+          for_body: [variable: ["i"]]
+        ]
+      )
+    end)
+  end
+
+  test "for tag: range with variables" do
+    test_combinator(
+      "{% for i in (my_var..10) %}{{ i }}{% endfor %}",
+      &Parser.for/1,
+      for: [
+        for_collection: [
+          variable_name: "i",
+          range_value: [start: "my_var", end: 10]
```

Comment:
I would like to be consistent and only use the key `value` since that is also what is used for variables for example. so here the key would be `value: {:range, [start: {:variable, "my_var"}, end: 10]}`

Reply:
You're, it looks better that way. It's more consistent and logical. I'm going to change it ASAP